Neos.ContentRepository.NodeAccess/Classes/FlowQueryOperations/UniqueOperation.php

```diff
@@ -0,0 +1,73 @@
+<?php
+namespace Neos\ContentRepository\NodeAccess\FlowQueryOperations;
+
+/*
+ * This file is part of the Neos.ContentRepository package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\ContentRepository\Core\Projection\ContentGraph\Filter\FindReferencesFilter;
+use Neos\ContentRepository\Core\Projection\ContentGraph\Node;
+use Neos\ContentRepositoryRegistry\ContentRepositoryRegistry;
+use Neos\Eel\FlowQuery\FlowQuery;
+use Neos\Eel\FlowQuery\OperationInterface;
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * "unique" operation working on Nodes
+ *
+ * This operation can be used to ensure that nodes are only once in the flow query context
+ *
+ *     ${q(node).backReferences().nodes().unique()get()}
+ *
+ * The result is an array of {@see Node} instances.
+ *
+ * @api To be used in Fusion, in php this should be implemented directly
+ */
+final class UniqueOperation implements OperationInterface
```

Comment:
this operation doesnt have to be node only. I think we could just check if `equals` exist in an object an compare objects that way. For simple types i would fallback to array_unique

Reply:
There should be a generic `unique` operation in eel itself that works by applying array_unique. That one will be for the flow-dev-collection as eel and basal flow query lives there. Will create that pr soonish but that should not stop this one here.